pkg/component/component.go

```
@@ -391,17 +430,28 @@ func Build(client *occlient.Client, componentName string, applicationName string
 	if err != nil {
 		return errors.Wrapf(err, "unable to rebuild %s", componentName)
 	}
-	if streamLogs {
-		if err := client.FollowBuildLog(buildName, stdout); err != nil {
-			return errors.Wrapf(err, "unable to follow logs for %s", buildName)
-		}
+
+	// Retrieve the Build Log and write to buffer if debug is disabled, else we we output to stdout / debug.
+
+	var b bytes.Buffer
+	if !log.IsDebug() {
+		stdout = bufio.NewWriter(&b)
 	}
+
+	if err := client.FollowBuildLog(buildName, stdout); err != nil {
+		return errors.Wrapf(err, "unable to follow logs for %s", buildName)
+	}
+
 	if wait {
 		if err := client.WaitForBuildToFinish(buildName); err != nil {
+			if !log.IsDebug() {
+				log.Errorf(b.String())
```

Maybe return this error instead of logging it here?